source/_integrations/unifiprotect.markdown

```diff
@@ -212,6 +213,35 @@ Your main UniFi Protect NVR device also gets a number of diagnostics sensors tha
 * **Disk Health**: Each disk installed in your NVR will have a disk health sensor. These are simple good/bad sensors and the order is not promised to match the order in UniFi OS. Disk model number is provided as a state attribute though to help map sensor to disk.
 * **Utilization and Storage Sensors**: Several other sensors are also added for uptime, hardware utilization, and distribution details of the video on disk.
 
+## Media Source
+
+A media source is provided for your UniFi Protect cameras so you can fetch events and video clips and thumbnails.
+
+### Media Browser
+
+The media source is split into 5 folders/levels:
+
+1. NVR Console Selector - only appears if you have more then one Protect NVR Console. Allows you to select your NVR Console you want to view events for.
+2. Camera Selector - either lets you select all cameras or a specific camera to view events for.
+3. Event Selector - either lets you select all events or a specific event type to view events for.
+4. Time Selector - filters events for a given time range:
+   * Last 24 Hours
+   * Last 7 Days
+   * Last 30 Days
+   * By Month since start of recording - selecting a month lets you either view the whole month or a specific date
+5. Event Selector - lets you select the specific event for playback
+
+Since the media browser does not really have any pagination or filtering built in all of the events must be loaded into memory. AS a result, the number of events loaded at once is truncated to 10,000 by default. The number of events will be listed at "10000 (TRUNCATED)" if the event count was truncated. You can raise or lower the limit of the number of events that can be loaded using the Config Entry Options.
```

Comment:
```suggestion
Since the media browser does not have any pagination or filtering, all of the events must be loaded into memory. As a result, the number of events loaded at once is truncated to 10,000 by default. The number of events will be listed at "10000 (TRUNCATED)" if the event count was truncated. You can raise or lower the limit of the number of events that can be loaded using the Config Entry Options.
```